### PR TITLE
docs: move docs to docs/dev folder

### PR DESCRIPTION
**Pull request by dgarcia360**
In order to allow our Scylla OSS customers the ability to select a version for their documentation, we are migrating the Scylla docs content to the Scylla OSS repository. This PR covers the following points of the [Migration Plan](https://docs.google.com/document/d/15yBf39j15hgUVvjeuGR4MCbYeArqZrO1ir-z_1Urc6A/edit#):

1. Creates a subdirectory for dev docs: /docs/dev
2. Moves the existing dev doc content in the scylla repo to /docs/dev, but keep Alternator docs in /docs.
3. Flattens the structure in /docs/dev (remove the subfolders).
4. Adds redirects from `scylla.docs.scylladb.com/<version>/<document>` to `https://github.com/scylladb/scylla/blob/master/docs/dev/<document>.md`
5. Excludes publishing docs for /docs/devs.

## How to test this PR

1. Enter the docs folder with `cd docs`.
2. Run `make redirects`.
3. Enter the docs folder and run `make preview`. The docs should build without warnings.
4. Open http://127.0.0.1:5500 in your browser. You shoul donly see the alternator docs.
5. Open http://127.0.0.1:5500/stable/design-notes/IDL.html in your browser. It should redirect you to https://github.com/scylladb/scylla/blob/master/docs/dev/IDL.md and raise a 404 error since this PR is not merged yet.
6. Surf the `docs/dev` folder. It should have all the scylla project internal docs without subdirectories.

### CODE REVIEW

**Comment by dgarcia360** [comment]
@tzach @annastuchlik Ready for review

File: README.md [hunk 1]
```diff
@@ -65,20 +65,20 @@ $ ./tools/toolchain/dbuild ./build/release/scylla --help
...
 
 ## Scylla APIs and compatibility
 By default, Scylla is compatible with Apache Cassandra and its APIs - CQL and
 Thrift. There is also support for the API of Amazon DynamoDB™,
 which needs to be enabled and configured in order to be used. For more
 information on how to enable the DynamoDB™ API in Scylla,
 and the current compatibility of this feature as well as Scylla-specific extensions, see
-[Alternator](docs/alternator/alternator.md) and
-[Getting started with Alternator](docs/alternator/getting-started.md).
+[Alternator](docs/source/alternator/alternator.md) and
```
Comment (nyh) [inline README.md:76]: What is this "source" subdirectory? I thought you said Alternator will be docs/alternator?

File: docs/Makefile [hunk 2]
```diff
@@ -69,7 +67,7 @@ epub3: setup
 
 .PHONY: multiversion
 multiversion: setup
-	$(POETRY) run sphinx-multiversion $(SOURCEDIR) $(BUILDDIR)/dirhtml
+	$(POETRY) run sphinx-multiversion source $(BUILDDIR)/dirhtml
```
Comment (nyh) [inline docs/Makefile:70]: Here's this "source" again. What is it, and why SOURCEDIR changed to "source" in this line, but a few lines below it's not changed?

File: docs/_utils/redirects.yaml [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+/stable/guides/building.html: https://github.com/scylladb/scylla/blob/master/docs/dev/building.md
+/stable/guides/debugging.html: https://github.com/scylladb/scylla/blob/master/docs/dev/debugging.md
+/stable/guides/docker-hub.html: https://github.com/scylladb/scylla/blob/master/docs/dev/docker-hub.md
+/stable/guides/logging.html: https://github.com/scylladb/scylla/blob/master/docs/dev/logging.md
+/stable/guides/testing.html: https://github.com/scylladb/scylla/blob/master/docs/dev/testing.md
+/stable/guides/tracing.html: https://github.com/scylladb/scylla/blob/master/docs/dev/tracing.md
+/stable/service_levels.html: https://github.com/scylladb/scylla/blob/master/docs/dev/service_levels.md
+
+# 4.6 branch
+/branch-4.6/contribute/index.html: https://github.com/scylladb/scylla/blob/master/docs/dev/index.md
```
Comment (nyh) [inline docs/_utils/redirects.yaml:45]: This redirects from branch 4.6 to master. Is this what we want to do?

**Review by nyh — COMMENTED** [review]
Hi, please see my comments below. Among other things, I didn't understand why alternator goes in a new subdirectory docs/source/alternator (what does "source" mean and why?).

File: README.md (hunk 1 above)
Comment (annastuchlik) [reply README.md:76]: @nyh The structure is the same as in other projects that include the docs, for example:
scylla-cloud-doc/docs/source/
java-driver/docs/source/
BUT we've just discussed it and David is going to push another commit to remove the "source" folder from the  structure.

Comment (dgarcia360) [reply README.md:76]: @nyh @annastuchlik Commit pushed to remove the source folder.

File: docs/_utils/redirects.yaml (hunk 3 above)
Comment (dgarcia360) [reply docs/_utils/redirects.yaml]: Good catch! I'm redirecting now to the specific branch.

File: docs/conf.py [hunk 4]
```diff
@@ -1,150 +1,134 @@
...
-master_doc = 'contents'
+master_doc = "index"
 
 # General information about the project.
-project = 'Scylla Dev'
-copyright = str(date.today().year) + ', ScyllaDB. All rights reserved.'
-author = u'Scylla Project Contributors'
+project = "ScyllaDB Documentation"
+copyright = str(date.today().year) + ", ScyllaDB. All rights reserved."
+author = u"Scylla Project Contributors"
```
Comment (annastuchlik) [inline docs/conf.py:48]:
```suggestion
author = u"ScyllaDB Project Contributors"
```

File: docs/dev/README.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+# Scylla developer documentation
+
+This folder (and its subfolders) contain developer-oriented documentation
```
Comment (annastuchlik) [inline docs/dev/README.md:3]:
```suggestion
This folder contains developer-oriented documentation
```

File: docs/dev/protocols.md [hunk 6]
```diff
@@ -197,7 +197,7 @@ HTTP (unencrypted) or HTTPS (encrypted) protocol. Support for this protocol
 is not turned on by default, and must be turned on manually by setting
 the `alternator_port` and/or `alternator_https_port` configuration options.
 "Alternator" is the codename of Scylla's DynamoDB API support, and is
-documented in more detail in [alternator.md](../alternator/alternator.md).
+documented in more detail in [alternator.md](../source/alternator/alternator.md).
```
Comment (annastuchlik) [inline docs/dev/protocols.md:200]: I think the "source" subfolder should be removed, following the changes.

**Review by annastuchlik — CHANGES_REQUESTED** [review]
I've left some comments. Most of them are minor. The important ones are about the project name and removing "source" from a path specified in one file.

File: docs/conf.py [hunk 7]
```diff
@@ -1,150 +1,134 @@
...
 
 # The master toctree document.
-master_doc = 'contents'
+master_doc = "index"
 
 # General information about the project.
-project = 'Scylla Dev'
-copyright = str(date.today().year) + ', ScyllaDB. All rights reserved.'
-author = u'Scylla Project Contributors'
+project = "ScyllaDB Documentation"
```
[the remark this answers is not in the document]
Comment (dgarcia360) [reply docs/conf.py:46]:
```suggestion
project = "ScyllaDB Open Source"
```

**Review by annastuchlik — APPROVED** [review]
LGTM

**Pull request closed without merging**